nixos/modules/services/misc/nix-gc.nix

```diff
@@ -90,7 +90,7 @@ in
 
     systemd.services.nix-gc = lib.mkIf config.nix.enable {
       description = "Nix Garbage Collector";
-      script = "exec ${config.nix.package.out}/bin/nix-collect-garbage ${cfg.options}";
+      serviceConfig.ExecStart = "${config.nix.package.out}/bin/nix-collect-garbage ${cfg.options}";
```

Comment:
nix-collect-garbage(1) has trivial flags so I see no danger here (yet), but this is another example of service arguments exposed to users without telling them how/where their arguments are parsed eventually.

So assuming https://github.com/NixOS/nixpkgs/pull/175907 goes in, this hunk could be
```
serviceConfig.ExecStart = "${config.nix.package.out}/bin/nix-collect-garbage "
  + lib.strings.escapeSystemdArgs cfg.options;
```